### PR TITLE
[IMP] Improve access right and groups and respected order of menu

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
- Task: https://www.odoo.com/web#id=7767&view_type=form&model=project.task&action=333&active_id=131
- Pad: https://pad.odoo.com/p/r.lFw6dVRQWxXWZUJk

### CODE REVIEW

File: addons/survey/security/survey_security.xml [hunk 1]
```diff
@@ -1,24 +1,13 @@
...
-            <field name="users" eval="[(4, ref('base.user_root'))]"/>
-        </record>
 
         <!-- Record rules -->
         <record id="survey_users_access" model="ir.rule">
             <field name="name">Access to survey for regular users</field>
             <field name="model_id" ref="survey.model_survey_survey"/>
             <field name="domain_force">[('stage_id.closed', '=', False)]</field>
-            <field name="groups" eval="[(4, ref('base.group_survey_user'))]"/>
+            <field name="groups" eval="[(4, ref('base.group_marketing_user'))]"/>
```
Comment (rim-odoo) [inline addons/survey/security/survey_security.xml]: Incorrect, anyone can answer a survey, even people that do not use marketing module at all

File: addons/marketing/security/marketing_security.xml [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
     </record>
 
-    <record id="marketing.group_marketing_manager" model="res.groups">
+    <record id="base.group_marketing_manager" model="res.groups">
         <field name="name">Manager</field>
         <field name="category_id" ref="base.module_category_marketing"/>
-        <field name="implied_ids" eval="[(4, ref('marketing.group_marketing_user'))]"/>
+        <field name="implied_ids" eval="[(4, ref('base.group_marketing_user'))]"/>
         <field name="users" eval="[(4, ref('base.user_root'))]"/>
     </record>
```
Comment (ara-odoo) [inline addons/marketing/security/marketing_security.xml]: remove this group and use administration group.

File: addons/gamification/security/gamification_security.xml [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
-            <field name="name">Manager</field>
-            <field name="category_id" ref="module_goal_category"/>
+    
+        <record id="base.group_hr_manager" model="res.groups">
+        <field name="name">Manager</field>
+            <field name="comment">the user will have an access to the human resources configuration as well as statistic reports.</field>
+            <field name="category_id" ref="base.module_category_human_resources"/>
+            <field name="implied_ids" eval="[(4, ref('base.group_user'))]"/>
             <field name="users" eval="[(4, ref('base.user_root'))]"/>
         </record>
```
Comment (ara-odoo) [inline addons/gamification/security/gamification_security.xml]: remove this group and use administration group.

File: addons/gamification/security/gamification_security.xml [hunk 4]
```diff
@@ -31,7 +29,7 @@
         <record id="goal_gamification_manager_visibility" model="ir.rule">
             <field name="name">Gamification Manager can see any goal</field>
```
Comment (ara-odoo) [inline addons/gamification/security/gamification_security.xml]: Rename to administration group.

File: addons/gamification/security/gamification_security.xml [hunk 5]
```diff
@@ -31,7 +29,7 @@
...
             <field name="model_id" ref="model_gamification_goal"/>
-            <field name="groups" eval="[(4, ref('group_goal_manager'))]"/>
+            <field name="groups" eval="[(4, ref('base.group_hr_manager'))]"/>
```
Comment (ara-odoo) [inline addons/gamification/security/gamification_security.xml]: use administrator `setting` group.

File: addons/gamification/security/gamification_security.xml [hunk 6]
```diff
@@ -1,14 +1,12 @@
...
-            <field name="name">Gamification</field>
-            <field name="description"></field>
-            <field name="sequence">17</field>
-        </record>
-        <record id="group_goal_manager" model="res.groups">
-            <field name="name">Manager</field>
-            <field name="category_id" ref="module_goal_category"/>
+    
+        <record id="base.group_hr_manager" model="res.groups">
+        <field name="name">Manager</field>
```
Comment (ara-odoo) [inline addons/gamification/security/gamification_security.xml]: indent your code.

File: addons/marketing_campaign/security/marketing_campaign_security.xml [hunk 7]
```diff
@@ -1,6 +1,17 @@
 <?xml version="1.0"?>
 <openerp>
     <data>
+    	<record id="group_lead_automation_user" model="res.groups">
+	        <field name="name">User</field>
```
Comment (ara-odoo) [inline addons/marketing_campaign/security/marketing_campaign_security.xml]: indent your code.

File: addons/survey/views/survey_views.xml [hunk 8]
```diff
@@ -516,7 +516,8 @@
         <!-- MENU ELEMENTS -->
 
         <!-- Left menu categories-->
-        <menuitem name="Surveys" id="menu_surveys" parent="base.marketing_menu" sequence="25" groups="base.group_survey_user"/>
+        <menuitem name="Surveys" id="menu_surveys" parent="base.marketing_menu" sequence="25" groups="base.group_user"/>
+        <menuitem name="Configuration" id="base.menu_marketing_config_root" parent="base.marketing_menu" sequence="30" />
```
Comment (ara-odoo) [inline addons/survey/views/survey_views.xml]: what is the purpose of this menu?

File: addons/share/security/share_security.xml [hunk 9]
```diff
@@ -1,10 +1,12 @@
...
-        <record model="ir.module.category" id="module_category_share">
-            <field name="name">Sharing</field>
-            <field name="parent_id" ref="base.module_category_tools"/>
-            <field name="sequence">26</field>
+
+        <record id="group_share_user" model="res.groups">
+            <field name="name">Share Documents</field>
+            <field name="comment">
+Members of this groups have access to the sharing wizard, which allows them to invite external users to view or edit some of their documents.</field>
+            <field name="category_id" ref="base.module_category_hidden"/>
```
Comment (ara-odoo) [inline addons/share/security/share_security.xml]: why you add this group?

File: addons/survey/views/survey_views.xml (hunk 8 above)
Comment (rim-odoo) [inline addons/survey/views/survey_views.xml]: Wasn't there already a "configuration" menu for marketing in Technical Features?

**Comment by ara-odoo** [comment]
maintain two commits one for sequence and another for security.

File: addons/survey/views/survey_views.xml (hunk 8 above)
Comment (gan-odoo) [inline addons/survey/views/survey_views.xml]: Yes, I remove It.

**Comment by rim-odoo** [comment]
Done in https://github.com/odoo/odoo/pull/7445

**Pull request closed without merging**